crates/crypto_api/src/buffer/mod.rs

```diff
@@ -73,6 +73,30 @@ pub trait Buffer: Send + std::fmt::Debug + Deref<Target = [u8]> + DerefMut<Targe
         }
         Ok(())
     }
+
+    /// compare this buffer to another buffer
+    /// Return :
+    /// | if a > b; return 1
+    /// | if a < b; return -1
+    /// | if a == b; return 0
+    #[allow(clippy::borrowed_box)]
+    fn compare(&mut self, b: &mut Box<dyn Buffer>) -> i32 {
```

Comment:
see other comment, this should be immutable
```suggestion
    fn compare(&self, b: &Box<dyn Buffer>) -> i32 {
```